cluster/saltbase/salt/kubelet/default

```diff
@@ -151,6 +151,8 @@
   {% set network_plugin = "--network-plugin=opencontrail" %}
 {% elif pillar.get('network_provider', '').lower() == 'cni' %}
   {% set network_plugin = "--network-plugin=cni --network-plugin-dir=/etc/cni/net.d/" %}
+{%elif pillar.get('policy_provider', '').lower() == 'calico' and grains['roles'][0] != 'kubernetes-master' -%}
```

Comment:
Hmm...

Yes, Calico currently needs information that it learns through CNI execution (e.g IP address, host-side veth, mac).  It's also important that Calico has a say at CNI execution time to prevent timing windows w.r.t security rules.

In canal, we (currently) are able to chain the flannel -> Calico -> host-local CNI plugins.  I'm afraid I haven't taken a super close look at kubenet yet - is this something that could be possible?